### PR TITLE
feat(observability): add HTTP metrics middleware and SLI/SLO dashboard

### PR DESCRIPTION
**Pull request by psincraian**
## Summary

Implement comprehensive HTTP request metrics tracking and SLI/SLO monitoring for the Polar API server with a Grafana dashboard for visibility into availability and latency metrics.

## What Changed

- **HTTP Metrics Middleware**: ASGI middleware that tracks request count and duration for all endpoints (except health checks)
- **Prometheus Remote Write**: Background pusher that sends metrics to Grafana Cloud
- **Grafana Dashboard**: Global + per-endpoint SLO views with error budget tracking and environment filtering
- **Comprehensive Tests**: 37 tests covering path normalization, metric collection, and edge cases

## Why This Matters

Enables SLI/SLO monitoring for checkout endpoints (99.95% availability, p99 latency targets) and provides a foundation for monitoring other API endpoints. Simplifies SLO definition by storing hardcoded thresholds in dashboard queries rather than Prometheus.

## How It Works

1. Middleware normalizes paths (UUIDs, numeric IDs, tokens) to prevent cardinality explosion
2. Metrics are collected in multiprocess-safe manner across API server instances
3. Remote write pusher batches metrics and sends to Grafana Cloud every 60 seconds
4. Dashboard filters by environment and allows drill-down to individual endpoints

## Testing

- [x] All existing tests pass
- [x] 37 new observability tests passing (path normalization, deny list, ASGI behavior)
- [x] Lint and type checking pass
- [x] Manual testing with metrics collection in local development

### CODE REVIEW

File: server/polar/observability/http_middleware.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+        # Primary: Use FastAPI's route object (most reliable)
+        # This is populated after routing completes, which is why we
+        # call this in the finally block after the request
+        route = scope.get("route")
+        if route and hasattr(route, "path"):
+            return route.path  # e.g., "/v1/checkouts/{id}"
+
+        # Fallback: Regex normalization for unmatched routes (404s, etc.)
+        return self._normalize_path(path)
```
Comment (psincraian) [inline server/polar/observability/http_middleware.py]: I don't want unkown routes to be tracked, this can be exploited
